test/extensions_test.dart

```diff
@@ -1352,6 +1353,122 @@ void main() {
         expect(l3.toList(), [4, 5]);
       });
     });
+    group('get frequencies tests', () {
+      test('should return correct frequency map for List of integers', () {
+        var list = [1, 2, 2, 3, 3, 3];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {1: 1, 2: 2, 3: 3});
+      });
+
+      test('should return correct frequency map for List of strings', () {
+        var list = ['a', 'b', 'b', 'c', 'c', 'c'];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {'a': 1, 'b': 2, 'c': 3});
+      });
+
+      test('should handle empty List', () {
+        var list = [];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {});
+      });
+
+      test('should handle single element List', () {
+        var list = [42];
+        var frequencyMap = list.frequencies;
+        expect(frequencyMap, {42: 1});
+      });
+
+      test('should return correct frequency map for Set of integers', () {
+        // ignore: equal_elements_in_set
+        var set = {1, 2, 2, 3, 3, 3};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {1: 1, 2: 1, 3: 1});
+      });
+
+      test('should return correct frequency map for Set of strings', () {
+        // ignore: equal_elements_in_set
+        var set = {'a', 'b', 'b', 'c', 'c', 'c'};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {'a': 1, 'b': 1, 'c': 1});
+      });
+
+      test('should handle empty Set', () {
+        var set = <int>{};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {});
+      });
+
+      test('should handle single element Set', () {
+        var set = {42};
+        var frequencyMap = set.frequencies;
+        expect(frequencyMap, {42: 1});
+      });
+
+      test('should return correct frequency map for Queue of integers', () {
+        var queue = Queue<int>();
+        queue.addAll([1, 2, 2, 3, 3, 3]);
+        var frequencyMap = queue.frequencies;
+        expect(frequencyMap, {1: 1, 2: 2, 3: 3});
+      });
+
+      test('should return correct frequency map for Queue of strings', () {
+        var queue = Queue<String>();
+        queue.addAll(['a', 'b', 'b', 'c', 'c', 'c']);
+        var frequencyMap = queue.frequencies;
+        expect(frequencyMap, {'a': 1, 'b': 2, 'c': 3});
+      });
+
+      test('should handle empty Queue', () {
+        var queue = Queue<int>();
+        var frequencyMap = queue.frequencies;
+        expect(frequencyMap, {});
+      });
+
+      test('should handle single element Queue', () {
+        var queue = Queue<int>();
+        queue.add(42);
+        var frequencyMap = queue.frequencies;
+        expect(frequencyMap, {42: 1});
+      });
+    });
+
+    group('get frequencies tests extended', () {
+      test('list of equal but not identical strings', () {
+        var list = ['apple', String.fromCharCodes('apple'.codeUnits)];
```

Comment:
String is not a good choice for this test because those will be identical when compiled for web.

Consider adding a class:
```dart
class Wrap {
  final int id;
  final int value;
  Wrap(this.id, this.value);
  int get hashCode => value.hashCode;
  bool operator ==(Object other) => other is Wrap && value == other.value;
  String toString() => '(#$id:$value)';
}
```
and using that to create distinct-but-equal values:
```dart
  var list = [Wrap(0, 1), Wrap(1, 1), Wrap(0, 2)];
  var frequenceyMap = list.frequencies;
  expect(frequencyMap[list[0]], 2);
  expect(frequencyMap[list[1]], 2);
  expect(frequencyMap[list[2]], 1);
```
or similar.